packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,234 @@
+
+Options:
+
+1. Find translations of each of our unit labels ("byte", "bytes", "KB", "MB", "GB", "TB", "PB", "KiB", "MiB", "GiB", "TiB", "PiB") for a fixed set of languages we wish to support. Maintain a mapping of language codes (e.g. "fr", "de", "zh_CN") to arrays of those localized unit labels. Given a locale to use, look up its language subtag in our map. If not found, fall back to English.
+
+**Pros:** No work for clients.
+
+**Cons:** Only supports a fixed set of languages. Up to us to find accurate translations. Unsure if subtags other than the language (e.g. region) could be relevant to the translation.
+
+2. Use Nimble label provider. Add label tokens for the units, and rely on clients to provide translations.
+
+**Pros:** Supports any language a client cares to provide translations for.
+
+**Cons:** Less convenient for clients. Does not honor the `lang` setting on the page or on `nimble-theme-provider`.
+
+3. Use `Intl.NumberFormat` and drop support for units based on a factor of 1024.
+
+**Pros:** No translation work for the client or Nimble team. Honors `lang` value.
+
+**Cons:** No support for 1024-based units. Shows "byte" unit even for multiple bytes. Missing several translations for "byte". Unclear if missing translations for other units (KB, MB, etc.).
+
+4. Use `Intl.NumberFormat` and support both 1024 or 1000 based conversions, but just always use KB/MB/GB/TB/PB as the units. There is precedent for this, so it may not be surprising to clients. This is what SLE's current implementation does.
+
+**Pros:** No translation work for the client or Nimble team. Honors `lang` value.
+
+**Cons:** Potentially misleading units label for 1024-based values. Shows "byte" unit even for multiple bytes. Missing several translations for "byte". Unclear if missing translations for other units (KB, MB, etc.).
+
+I suggest we go with option 2, because it results in the best unit labels, without giving up any functionality, and with translation accuracy being the concern of the client.
+
+### Security
+
+N/A
+
+### Performance
+
+N/A
+
+### Dependencies
+
+None
+
+### Test Plan
+
+Unit tests:
+
+-   standard component tests
+-   renders blank for invalid input values: `Inf`, `-Inf`, `NaN`, and non-number
+-   for an interesting range of values, value conversion results in correct values and units (using English `lang`)
+-   correct units used for both `unit-type`s
+-   standard cell text/group header text ellipsizing tests
+-   honors `lang` value and responds to changes to `lang` value
+-   uses English labels when unsupported locale is given
+
+### Tooling
+
+N/A
+
+### Documentation
+
+This component will be documented via a new story in Storybook.
+
+---
+
+## Open Issues
+
+-   Is there a better name for this column type? E.g. `nimble-table-column-memory-size`.
+-   Should we support grouping?
+-   What is our unit localization approach?
```

Comment:
I think another open question related to unit localization is what the expected/acceptable behavior is from our POs. Some of the tradeoffs for localization have to do with showing `MB` rather than `MiB`, and we probably need someone to decide if that's an acceptable tradeoff to make.